localstack/utils/aws/message_forwarding.py

```diff
@@ -166,20 +238,38 @@ def send_event_to_api_destination(target_arn, event):
         "Connection": "close",
     }
 
-    # add auth headers for target destination
-    add_api_destination_authorization(destination, headers, event)
+    connection_arn = destination.get("ConnectionArn", "")
```

Comment:
Let's put this block (lines 241-267) into `add_api_destination_authorization(..)` below - we may then need to have this function return the `endpoint` (as it can be updated and is required in here).

The main reason is that the `add_api_destination_authorization(..)` is already used in some plugin extensions, and should be kept here for overriding..